sql/sqlwire/http.go

```diff
@@ -17,8 +17,49 @@
 
 package sqlwire
 
+import (
+	gogoproto "github.com/gogo/protobuf/proto"
+)
+
 const (
 	// Endpoint is the URL path prefix which accepts incoming
 	// HTTP requests for the SQL API.
 	Endpoint = "/sql/"
 )
+
+// Request is an interface for RPC requests.
+type Request interface {
+	gogoproto.Message
+	Header() *SQLRequestHeader
+	// Method returns the request method.
+	Method() Method
+	// CreateReply creates a new response object.
+	CreateReply() Response
+}
+
+// Response is an interface for RPC responses.
+type Response interface {
+	gogoproto.Message
+	// Header returns the response header.
+	Header() *SQLResponseHeader
+}
+
+// Header returns the request header.
+func (r *SQLRequest) Header() *SQLRequestHeader {
```

Comment:
This is done slightly more cleverly for `proto.Request`. Since `SQLRequestHeader` is embedded in `SQLRequest`, the methods of `SQLRequestHeader` are "inherited" by `SQLRequest`. So we can define the `Header()` method on `SQLRequestHeader` and have it "inherited" to any message that embeds `SQLRequestHeader`:

```
func (h *SQLRequestHeader) Header() *SQLRequestHeader {
  return h
}
```

The same technique applies to `SQLResponse`.